src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -1154,23 +1182,36 @@ public synchronized ResultIterator iterator(Expression query) throws DatabaseEng
 
     @Override
     public ResultIterator iterator(String query, int fetchSize) throws DatabaseEngineException {
+        return iterator(query, fetchSize, properties.getSelectQueryTimeout());
+    }
+
+    @Override
+    public ResultIterator iterator(Expression query, int fetchSize) throws DatabaseEngineException {
+        return iterator(translate(query), fetchSize);
+    }
+
+    @Override
+    public ResultIterator iterator(Expression query, int fetchSize, int readTimeoutOverride) throws DatabaseEngineException {
+        return iterator(translate(query), fetchSize, readTimeoutOverride);
+    }
+
+    @Override
+    public ResultIterator iterator(String query, int fetchSize, int readTimeoutOverride) throws DatabaseEngineException {
```

Comment:
The first `iterator` method in AbstractDatabaseEngine had this signature:  
```
public synchronized ResultIterator iterator(String query)
```
Then people added more iterator methods, some of them `synchronized`, some of them not.
I think all methods that use the connection should be `synchronized`; this means this one would need to be `synchronized`, but all other `iterator` methods call this one and don't need to be.

Reply:
Yep, removed some unnecessary synchronized. Ensured the processResultIterator was kept synchronized because it iterates in the result set, and that might involve fetching rows from the server.